controllers/util/solr_util.go

```diff
@@ -520,8 +520,15 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	}
 
 	if solrCloud.Spec.SolrImage.ImagePullSecret != "" {
-		stateful.Spec.Template.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
-			{Name: solrCloud.Spec.SolrImage.ImagePullSecret},
+		if nil != customPodOptions {
+			stateful.Spec.Template.Spec.ImagePullSecrets = append(
+				customPodOptions.ImagePullSecrets,
```

Comment:
I think customPodOptions secrets will only be added if the `SolrImage.ImagePullSecret` is provided. You should probably have it added in a separate if statement.

Reply:
Fixed! Thanks for pointing that out